src/App.js

```diff
@@ -14,11 +17,113 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const preparedGoods = goodsFromServer.map((good, index) => (
+  {
+    name: good,
+    id: index,
+  }
+));
+
+class App extends React.Component {
+  state = {
+    isShown: false,
+    goods: preparedGoods,
+    selected: 1,
+  };
+
+  clickHandler = () => {
+    this.setState({
+      isShown: true,
+    });
+  }
+
+  reverseGoods = () => {
+    this.setState(prevState => (
+      {
+        goods: [...prevState.goods].reverse(),
+      }
+    ));
+  }
+
+  sortAlphabetically = () => {
+    this.setState(prevState => (
+      {
+        goods: [...prevState.goods].sort((x, y) => (
```

Comment:
принято называть это a, b
```suggestion
        goods: [...prevState.goods].sort((a, b) => (
```